### PR TITLE
Sensible Defaults:Builds

### PR DESCRIPTION
**Pull request by michaelongso**
This work is to setup a scheduled nightly build. This is an alignment with the [Sensible Defaults:Builds](https://github.com/OctopusDeploy/Architecture/blob/main/SensibleDefaults/Builds.md) architectural direction. (see: `Everything Builds Every Night`)

Taking this opportunity to improve GHA result by displaying unit test results.

### CODE REVIEW

File: build/Build.cs [hunk 1]
```diff
@@ -82,7 +82,9 @@ class Build : NukeBuild
                 .SetProjectFile(Solution)
                 .SetConfiguration(Configuration)
                 .SetNoBuild(true)
-                .EnableNoRestore());
+                .EnableNoRestore()
+                .SetLoggers("trx")
```
Comment (michaelongso) [inline build/Build.cs:86]: Output the TRX files to display GHA unit test results

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -48,13 +64,26 @@ jobs:
...
+      if: success() || failure()    # run this step even if previous step failed
+      with:
+        name: Linux unit test results
+        path: ./TestResults/*.trx
+        reporter: dotnet-trx
+        fail-on-error: true
+
     - name: Tag release (when not pre-release) 🏷️
-      if: ${{ !contains( steps.build.outputs.octoversion_fullsemver, '-' ) }}
+      if: ${{ github.event_name != 'schedule' && !contains( steps.build.outputs.octoversion_fullsemver, '-' ) }}
```
Comment (michaelongso) [inline .github/workflows/main.yml:86]: Making sure version TAG does not happen for scheduled build

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -48,13 +64,27 @@ jobs:
       uses: actions/setup-dotnet@v2
       with:
         dotnet-version: '6.0.x'
+      
+    # Adjustment is done prior to Nuke build as OCTOVERSION information is included in the result package.
+    - name: Append OCTOVERSION_CurrentBranch with -nightly (for scheduled)
+      if: github.event_name == 'schedule'
+      run: echo "OCTOVERSION_CurrentBranch=${{ env.OCTOVERSION_CurrentBranch }}-nightly" >> $GITHUB_ENV
```
Comment (michaelongso) [inline .github/workflows/main.yml:71]: For scheduled event, expecting `OCTOVERSION_CurrentBranch` = `refs/heads/main`
Adjustment = `refs/heads/main-nightly`
Resulting `OCTOVERSION_FullSemVer` major.minor.patch`-main-nightly`

File: .nuke/build.schema.json [hunk 4]
```diff
@@ -75,8 +75,6 @@
               "CalculateVersion",
               "Clean",
               "Compile",
-              "CopyToLocalPackages",
```
Comment (michaelongso) [inline .nuke/build.schema.json:78]: This file is updated from running the Nuke build command locally. The build steps were removed prior to this PR.

File: .github/workflows/main.yml (hunk 3 above)
Comment (skermajo) [reply .github/workflows/main.yml:71]: Having `main` show in versions is atypical, given it's the default branch.

Is there a reason to recalculate the existing branch name? Above it's already set to `${{ github.head_ref || github.ref }}`. Would it be safer to simply append `-nightly` to whatever the current value happens to be?

**Review by skermajo — COMMENTED** [review]
This is looking really good and well thought out, thankyou. Just one question around resetting the branch name.

File: .github/workflows/main.yml (hunk 3 above)
Comment (skermajo) [reply .github/workflows/main.yml:71]: Ok, sorry I understand now. I was confused by `-main` being in the pre-release pacakge name. This does indeed make sense, as `OctoVersion` now sees the branch name as `main-nightly`, which will not match its conditions for building a release version without the branch name in the pre-release tag. My mistake!

**Review by skermajo — APPROVED** [review]
This looks really good. Thanks for the thought that went into this. It looks like a solid foundation for our other nightly builds 👍

**Pull request merged**